app/src/main/java/com/example/giphyservice/ui/MainActivity.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.giphyservice.ui
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.giphyservice.R
+import com.example.giphyservice.data.model.Gif
+import com.example.giphyservice.ui.list.GifsAdapter
+
+const val BASE_URL = "https://api.giphy.com/v1/"
+
+//the View observes changes in the ViewModel and updates its state accordingly
+class MainActivity : AppCompatActivity() {
+    private var viewModel: MainViewModel = MainViewModel()
```

Comment:
You should create viewModel in another way. Please read documentation about viewModel:
https://developer.android.com/topic/libraries/architecture/viewmodel#java